packages/opencensus-core/src/trace/model/span.ts

```diff
@@ -13,60 +13,370 @@
+  /** Trace Parameters */
+  activeTraceParams: configTypes.TraceParams = {};
+
+  /** The number of dropped attributes. */
+  droppedAttributesCount = 0;
+  /** The number of dropped links. */
+  droppedLinksCount = 0;
+  /** The number of dropped annotations. */
+  droppedAnnotationsCount = 0;
+  /** The number of dropped message events. */
+  droppedMessageEventsCount = 0;
+
+  /** Constructs a new Span instance. */
+  constructor(parent?: Span) {
+    this.className = this.constructor.name;
+    this.id = randomSpanId();
+    this.spansLocal = [];
+    if (parent) {
+      this.root = parent.root;
+      this.parentSpan = parent;
+      this.activeTraceParams = this.root.activeTraceParams;
+    } else {
+      this.root = this;
+    }
+    this.logger = (this.root && this.root.logger) || this.logger;
   }
 
-  /** Gets trace id of span. */
+  /** Gets the trace ID. */
   get traceId(): string {
     return this.root.traceId;
   }
 
+  /** Gets the trace state */
   get traceState(): types.TraceState|undefined {
     return this.root.traceState;
   }
 
-  /** Starts the span instance. */
-  start() {
-    super.start();
-    this.logger.debug('starting span  %o', {
+  /**
+   * Gets the ID of the parent span.
+   * RootSpan doesn't have a parentSpan but it override this method.
+   */
+  get parentSpanId(): string {
+    if (!this.parentSpan) {
+      return 'no-parent';
+    }
+    return this.parentSpan.id;
+  }
+
+  /** Indicates if span was started. */
+  get started(): boolean {
+    return this.startedLocal;
+  }
+
+  /** Indicates if span was ended. */
+  get ended(): boolean {
+    return this.endedLocal;
+  }
+
+  /**
+   * Gives a timestamp that indicates the span's start time in RFC3339 UTC
+   * "Zulu" format.
+   */
+  get startTime(): Date {
+    if (!this.clock) {
+      this.logger.debug('calling startTime() on null clock');
+      return new Date();
+    }
+
+    return this.clock.startTime;
+  }
+
+  /** Recursively gets the descendant spans. */
+  allDescendants(): types.Span[] {
+    console.log('Calling allDescendants', this.name);
+    return this.spansLocal.reduce((acc: types.Span[], cur) => {
+      console.log('Name', cur.name, acc.length);
+      acc.push(cur);
+      const desc = cur.allDescendants();
+      acc = acc.concat(desc);
+      console.log(cur.name, acc.length);
+      return acc;
+    }, []);
+  }
+
+  /** The list of immediate child spans. */
+  get spans(): types.Span[] {
+    return this.spansLocal;
+  }
+
+  /** The number of direct children. */
+  get numberOfChildren(): number {
+    return this.spansLocal.length;
+  }
+
+  /**
+   * Gives a timestap that indicates the span's end time in RFC3339 UTC
+   * "Zulu" format.
+   */
+  get endTime(): Date {
+    if (!this.clock) {
+      this.logger.debug('calling endTime() on null clock');
+      return new Date();
+    }
+
+    return this.clock.endTime;
+  }
+
+  /**
+   * Gives a timestap that indicates the span's duration in RFC3339 UTC
```

Comment:
Instead -> Gets the duration of the clock.

Reply:
Done